tests/integrations/raftstore/test_lease_read.rs

```diff
@@ -376,21 +376,24 @@ fn test_read_index_when_transfer_leader() {
     let resp2 = read_on_old_leader!();
 
     // Unpark all pending messages and clear all filters.
-    let router = cluster.sim.wl().get_router(leader.get_id()).unwrap();
+    let router = cluster.sim.wl().get_router(old_leader.get_id()).unwrap();
     'LOOP: loop {
         for raft_msg in mem::replace(dropped_msgs.lock().unwrap().as_mut(), vec![]) {
-            if raft_msg.get_message().get_msg_type() == MessageType::MsgAppend {
-                // All heartbeat responses are received before the new leader's MsgAppend.
-                cluster.sim.wl().clear_recv_filters(leader.get_id());
-                break 'LOOP;
+            let msg_type = raft_msg.get_message().get_msg_type();
+            if msg_type == MessageType::MsgHeartbeatResponse {
+                router.send_raft_message(raft_msg).unwrap();
+                continue;
             }
-            router.send_raft_message(raft_msg).unwrap();
+            cluster.sim.wl().clear_recv_filters(old_leader.get_id());
+            break 'LOOP;
         }
     }
 
+    let resp1 = resp1.recv().unwrap();
+    assert_eq!(resp1.get_responses()[0].get_get().get_value(), b"v2");
+
     // Response 2 should contains an error.
-    drop(resp1.recv().unwrap());
     let resp2 = resp2.recv().unwrap();
-    assert!(resp2.get_header().get_error().has_not_leader());
+    assert!(resp2.get_header().get_error().has_stale_command(), "{:?}", resp2);
```

Comment:
Why it's not leader before, but stale command now?

Reply:
It should be stale command because the second read index will be cleared in `clear_uncommitted`.